lib/structs/events/voip.cpp

```diff
@@ -24,25 +24,54 @@ add_version(json &obj, std::string_view version)
 }
 }
 
-namespace mtx::events::msg {
+namespace mtx::events::voip {
+
+// RTC Session Description 
+void
+from_json(const json &obj, RTCSessionDescriptionInit &content)
+{
+    content.sdp  = obj.at("sdp").get<std::string>();
+    if (obj.at("type").get<std::string>() == "answer")
+        content.type = RTCSessionDescriptionInit::Type::Answer;
+    else if(obj.at("type").get<std::string>() == "offer")
+        content.type = RTCSessionDescriptionInit::Type::Offer;
+        
+}
+
+void
+to_json(json &obj, const RTCSessionDescriptionInit &content)
+{
+    obj["sdp"]  = content.sdp;
+    if (content.type == RTCSessionDescriptionInit::Type::Answer)
+        obj["type"] = "answer";
+    else if (content.type == RTCSessionDescriptionInit::Type::Offer)
+        obj["type"] = "offer";
+}
 
 // m.call.invite
 void
 from_json(const json &obj, CallInvite &content)
 {
     content.call_id  = obj.at("call_id").get<std::string>();
-    content.sdp      = obj.at("offer").at("sdp").get<std::string>();
+    content.offer      = obj.at("offer").get<RTCSessionDescriptionInit>();
     content.version  = version(obj);
     content.lifetime = obj.at("lifetime").get<uint32_t>();
+    if(content.version != "0")
+    {
+        content.party_id  = obj.at("party_id").get<std::string>();
+        content.invitee = obj.at("invitee").get<std::string>();
+    }
```

Comment:
The MSC says that version 0 is a number, everything else is a string.

Reply:
Oops, sorry, I misread your code as changing the version parsing, ignore my comment please :D